Improve robustness of PD algorithm

I was doing a test run with the simple pd algorithm on the server
```
ubuntu@master:/data/mathosphere/mathosphere-core/target$ /srv/flink/bin/flink run mathosphere-core-3.0.0-SNAPSHOT-jar-with-dependencies.jar pd --inputDir hdfs://m:54310/data/ntcir/12/arxiv/xhtml/together.xml --outputDir hdfs://m:54310/out/m1 --referenceDir hdfs://m:54310/data/ex1.html
```
and got the following exception
```
Caused by: java.lang.Exception: Serialized representation of javax.xml.xpath.XPathExpressionException: net.sf.saxon.trans.XPathException: Finding root of tree: the context item is absent
        at net.sf.saxon.xpath.XPathExpressionImpl.evaluate(XPathExpressionImpl.java:378)
        at com.formulasearchengine.mathmlquerygenerator.xmlhelper.XMLHelper.getElementsB(XMLHelper.java:184)
        at com.formulasearchengine.mathosphere.mathpd.pojos.ArxivDocument.getMathTags(ArxivDocument.java:33)
        at com.formulasearchengine.mathosphere.mathpd.pojos.ArxivDocument.getCElements(ArxivDocument.java:38)
        at com.formulasearchengine.mathosphere.mathpd.FlinkPd$1.reduce(FlinkPd.java:41)
        at org.apache.flink.runtime.operators.AllGroupReduceDriver.run(AllGroupReduceDriver.java:148)
        at org.apache.flink.runtime.operators.BatchTask.run(BatchTask.java:486)
        at org.apache.flink.runtime.operators.BatchTask.invoke(BatchTask.java:351)
        at org.apache.flink.runtime.taskmanager.Task.run(Task.java:584)
        at java.lang.Thread.run(Thread.java:745)
```

--- a/mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/FlinkPd.java
+++ b/mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/FlinkPd.java
@@ -8,7 +8,10 @@ import com.formulasearchengine.mathosphere.mlp.contracts.JsonSerializerMapper;
 import com.formulasearchengine.mathosphere.mlp.contracts.TextAnnotatorMapper;
 import com.formulasearchengine.mathosphere.mlp.pojos.ParsedWikiDocument;
 import com.formulasearchengine.mathosphere.mlp.pojos.WikiDocumentOutput;
+import com.formulasearchengine.mathosphere.mml.CMMLInfo;
 import com.google.common.collect.Multiset;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 import org.apache.flink.api.common.functions.GroupReduceFunction;
 import org.apache.flink.api.common.functions.ReduceFunction;
 import org.apache.flink.api.common.typeutils.base.array.StringArraySerializer;
@@ -21,7 +24,11 @@ import org.apache.flink.core.fs.FileSystem.WriteMode;
 import org.apache.flink.core.fs.Path;
 import org.apache.flink.util.Collector;
 
+import javax.xml.xpath.XPathExpressionException;
+import java.util.logging.Logger;
+
 public class FlinkPd {
+    protected static final Log LOG = LogFactory.getLog(FlinkPd.class);
 
     public static void main(String[] args) throws Exception {
         FlinkPdCommandConfig config = FlinkPdCommandConfig.from(args);
@@ -38,9 +45,13 @@ public class FlinkPd {
                     @Override
                     public void reduce(Iterable<Tuple2<ArxivDocument, ArxivDocument>> iterable, Collector<Tuple2<Integer, String>> collector) throws Exception {
                         for (Tuple2<ArxivDocument, ArxivDocument> i : iterable) {
-                            final Multiset<String> elements = i.f1.getCElements();
-                            elements.removeAll(i.f0.getCElements());
-                            collector.collect(new Tuple2(elements.size(), i.f1.title + "-" + i.f0.title));
+                            try {
+                                final Multiset<String> elements = i.f1.getCElements();
+                                elements.removeAll(i.f0.getCElements());
+                                collector.collect(new Tuple2(elements.size(), i.f1.title + "-" + i.f0.title));
+                            }catch( XPathExpressionException xPathExpressionException){
+                                LOG.error("could not parse document: "+i.f0.title + " OR "+i.f1.title, xPathExpressionException);
+                            }
                         }
                     }
                 })

